### PR TITLE
[LETS-126] - Include unit tests in the appveyor build

### PR DESCRIPTION
**Pull request by danilailie**
http://jira.cubrid.org/browse/LETS-126

* Fixed unit tests compile errors for windows.

### CODE REVIEW

File: unit_tests/log/dummy_page_server.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#include "log_replication.hpp"
```
Comment (ribram) [inline unit_tests/log/dummy_page_server.cpp:19]: Add license header.

File: unit_tests/log/CMakeLists.txt [hunk 2]
```diff
@@ -273,42 +279,43 @@ add_dependencies(test_data_page_fetcher
...
-  )
-target_compile_definitions(test_log_recovery_parallel PRIVATE
-  ${COMMON_DEFS}
-  SERVER_MODE
-  )
-target_include_directories(test_log_recovery_parallel PRIVATE
-  ${TEST_INCLUDES}
-  )
-#target_link_libraries(test_log_recovery_parallel PRIVATE test_common)
 if(UNIX)
```
Comment (ribram) [inline unit_tests/log/CMakeLists.txt:315]: Why are these unit tests limited to linux?

File: unit_tests/packing/CMakeLists.txt [hunk 3]
```diff
@@ -44,15 +44,15 @@ target_include_directories(test_packing PRIVATE
...
+    test_common
     cubrid
     )
 elseif(WIN32)
-	target_link_libraries(test_packing PRIVATE
+	target_link_libraries(test_packing PRIVATE 
+    test_common
     cubrid-win-lib
     )
 else()
```
Comment (ribram) [inline unit_tests/packing/CMakeLists.txt]: Is this change required? I think test_packing does not need fake_thread_utils.

File: unit_tests/log/CMakeLists.txt [hunk 4]
```diff
@@ -231,7 +236,7 @@ target_compile_definitions(test_log_lsa_utils PRIVATE
 target_include_directories(test_log_lsa_utils PRIVATE
   ${TEST_INCLUDES}
   )
-# target_link_libraries(test_log_lsa_utils fake_thread_utils)
+target_link_libraries(test_log_lsa_utils fake_thread_utils)
```
Comment (ribram) [inline unit_tests/log/CMakeLists.txt]: Does test_log_lsa_utils depend on fake_thread_util? It doesn't like it does. If it is does not, better to remove the line completely.

**Review by ribram — CHANGES_REQUESTED** [review]
The issue (and PR title) mentions including building unit tests in the appveyor build. Update .appveyor.yml.

File: unit_tests/log/ut_database.hpp [hunk 5]
```diff
@@ -97,10 +97,10 @@ class ut_database_values_generator
     std::uniform_real_distribution<double> m_duration_in_millis_dist;
 
     // discrete distribution for adding new volumes or updating existing volumes
-    std::discrete_distribution<int8_t> m_add_or_update_volume_dist;
+    std::discrete_distribution<int32_t> m_add_or_update_volume_dist;
 
     // discrete distribution for adding new pages or updating existing pages
-    std::discrete_distribution<int8_t> m_add_or_update_page_dist;
+    std::discrete_distribution<int32_t> m_add_or_update_page_dist;
```
Comment (ribram) [inline unit_tests/log/ut_database.hpp:103]: Why this change? Please explain.

**Comment by ribram** [comment]
@danilailie the appveyor build is still missing -DUNIT_TESTS option that triggers unit test compilation.

File: unit_tests/log/ut_database.hpp (hunk 5 above)
Comment (danilailie) [reply unit_tests/log/ut_database.hpp:103]: On vs compiler it complains that the template instantiation with an 8 bit type is illegal.

Comment (ribram) [reply unit_tests/log/ut_database.hpp:103]: Right, the standard documentation for [discrete_distribution](https://en.cppreference.com/w/cpp/numeric/random/discrete_distribution) mentions:

<html>
<body>
<h3 style="color: rgb(0, 0, 0); background: none rgb(255, 255, 255); font-weight: bold; margin: 0px 0px 0.3em; overflow: hidden; padding: 1.1em 0px 0.2em 0.75em; border-bottom: none; width: auto; font-size: 16.896px; font-family: DejaVuSans, &quot;DejaVu Sans&quot;, arial, sans-serif; font-style: normal; font-variant-ligatures: no-common-ligatures; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="mw-headline" id="Template_parameters">Template parameters</span></h3>

IntType | - | The result type generated by the generator. The effect is undefined if this is not one of short, int, long, long long, unsigned short, unsigned int, unsigned long, or unsigned long long.
-- | -- | --



</body>
</html>

Shouldn't m_add_or_update_page_dist & m_add_or_update_page_dist usages be updated too, as consequence of this change?

**Comment by ribram** [comment]
Please check appveyor error:
```
CMake Error: Could not create named generator Visual Studio 16 2019 Win64
CMake Error: CMAKE_GENERATOR was set but the specified generator doesn't exist. Using CMake default.
Generators
* Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
  Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
```
The fix should be adding `-A` before Win64.

File: unit_tests/log/ut_database.hpp (hunk 5 above)
Comment (danilailie) [reply unit_tests/log/ut_database.hpp:103]: It's done

**Review by ribram — APPROVED** [review]

**Pull request merged**